### DISCUSSION TITLE
DeepgramClient ignoring API key

### DISCUSSION THREAD

**Discussion by dunctk** [deepgram/community#577] [General help] [answered]
### Which Deepgram product are you using?

Deepgram SDKs

### Details

I'm trying to use DeepgramClient as described in the documentation, however it won't accept my API key. It seems to say that it's blank. 

I added a step to print the length of they key so I can verify it's not empty. 

`config = DeepgramClientOptions(
            verbose=logging.SPAM,
        )

print(f'len of deepgram key={len(DEEPGRAM_KEY)}')
deepgram = DeepgramClient(api_key=DEEPGRAM_KEY, config=config)`

This prints:

`len of deepgram key=40
WARNING: API key is missing`

then later:

`deepgram.clients.errors.DeepgramApiError: DeepgramApiError: Invalid credentials. (Status: 401)`

[According to the docs](https://developers.deepgram.com/docs/python-sdk-pre-recorded-transcription) here this is the correct way to pass the key. Is that incorrect?

I'm using version 3.1.1

### If you are making a request to the Deepgram API, what is the full Deepgram URL you are making a request to?

deepgram.listen.prerecorded.v("1").transcribe_file(payload, options)

### If possible, please attach your code or paste it into the text box.

import os
from dotenv import load_dotenv
import logging, verboselogs
from deepgram import (
    DeepgramClient,
    DeepgramClientOptions,
    PrerecordedOptions,
    FileSource,
)

DEEPGRAM_KEY = os.getenv('DEEPGRAM_KEY')

def transcribe_video(video_file_path):
    print('transcribing reference video...')

    config = DeepgramClientOptions(
            verbose=logging.SPAM,
        )

    print(f'len of deepgram key={len(DEEPGRAM_KEY)}')
    deepgram = DeepgramClient(api_key=DEEPGRAM_KEY, config=config)

    with open(video_file_path, "rb") as file:
        buffer_data = file.read()

    payload: FileSource = {
            "buffer": buffer_data,
        }

    options = PrerecordedOptions(
        model="nova",
        smart_format=True,
        utterances=True,
        punctuate=True,
        diarize=True,
    )

    response = deepgram.listen.prerecorded.v("1").transcribe_file(payload, options)

    print(response)

    return response

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Reply by dunctk** [reply]
I removed the config and just used `deepgram = DeepgramClient(DEEPGRAM_API_KEY)` and it works now. It seems like it ignores the api key if config is also present.

**Reply by lukeocodes** [reply]
Would it not be correct to provide it like so `deepgram = DeepgramClient(DEEPGRAM_API_KEY, config)`?

If you believe there is a bug in the SDK, I would encourage you to [raise it as an issue on the SDK](https://github.com/deepgram/deepgram-python-sdk) itself.

**Reply by dunctk** [reply]
I normally prefer to use argument names as I find it's more explicit and works better for readability and clarity. Either way it seems like adding the config argument stops it reading the API key somehow. Maybe it's expecting the key within config?

I don't need the config, I just copied it from the docs. I've removed it and it works fine.

**Comment by jkroll-deepgram** [comment, marked as answer]
I was able to replicate this. Opened a new Python SDK issue: https://github.com/deepgram/deepgram-python-sdk/issues/283

As @dunctk also found, it worked for me to omit the `config` parameter in the meantime.

**Comment by davidvonthenen** [comment]
@dunctk This should be fixed and available in this release:
https://github.com/deepgram/deepgram-python-sdk/releases/tag/v3.1.2

**Reply by dunctk** [reply]
Thanks!
